### DISCUSSION TITLE
blog/refresh-tokens-in-aspnet-core/

### DISCUSSION THREAD

**Discussion by giscus** [codewithmukesh/discussions#16] [General]
# blog/refresh-tokens-in-aspnet-core/

In our previous article, we learned about Securing ASP.NET Core API with JWT Authentication. Now, let's go through Refresh Tokens in ASP.NET Core APIs that use JWT Authentication. We'll be using the codebase that we built in the previous article and add functionalities that support Refreshing JWT Tokens.

https://codewithmukesh.com/blog/refresh-tokens-in-aspnet-core/

**Comment by kokilasanjeewa** [comment]
great

**Comment by Mandill** [comment]
Veey good blog

**Comment by Aathi-Narayanan** [comment]
Can we use api for mobile development set token in cookies is good practice?

**Comment by shubham-kelvin** [comment]
if (!exisitngRefreshToken.isActive)
   {
       if (exisitngRefreshToken.Revoked is not null)
       {
           await RevokeAllExistingToken(exisitngRefreshToken.UserId, context);
       throw new Exception("Token is stolen revoked all tokens");
       }

// missed case
       if (DateTime.UtcNow >= exisitngRefreshToken.ExpiresAt)
       {
           // expired token case
           throw new Exception("Refresh token expired");
       }
   }
Expired refresh tokens were being treated as stolen tokens.
you missed this case
